include object's own frame

Licini opened the pull request:
@tomvanmele I know we had some discussion on moving responsibility of handling transformation out of Scene. But before we actually do that, there was a mistake here to fix. The `worldtransformation` was not including the object's own frame. This PR is to fix that.

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

Licini committed:
include object's own frame

tomvanmele reviewed (approved):
LGTM. although would be nice if there were an example in the docs :)

Licini committed:
update docs and add test

Licini commented:
> LGTM. although would be nice if there were an example in the docs :)

Just updated the relevant section in the docs. Also added a test. btw what should I do to pass this codecov test??
